mods/default/tools.lua

```diff
@@ -330,3 +330,70 @@ minetest.register_tool("default:sword_diamond", {
 		damage_groups = {fleshy=8},
 	}
 })
+
+minetest.register_tool("default:skeleton_key", {
+	description = "Skeleton Key",
+	inventory_image = "default_key_skeleton.png",
+	groups = {key = 1},
+	on_place = function(itemstack, placer, pointed_thing)
+		if pointed_thing.type ~= "node" then
+			return nil
+		end
+
+		local pos = pointed_thing.under
+		local node = minetest.get_node(pos)
+
+		if not node or node.name == "ignore" then
+			return nil
+		end
+
+		local on_skeleton_key_use = minetest.registered_nodes[node.name].on_skeleton_key_use
+		if on_skeleton_key_use then
+			-- make a new key secret in case the node callback needs it
+			local newsecret = string.format("%04x", math.random(65535))
+			newsecret = newsecret .. string.format("%04x", math.random(65535))
+			newsecret = newsecret .. string.format("%04x", math.random(65535))
+			newsecret = newsecret .. string.format("%04x", math.random(65535))
```

Comment:
Single `format` might be easier to read and should be easier on memory.
```lua
local random = math.random
local newsecret = string.format(
	"%04x%04x%04x%04x",
	 random(2^16)-1, random(2^16)-1, random(2^16)-1, random(2^16)-1
)
```
`random` starts from 1, but maybe `random(0, 2^16-1)` would work, too.